lib/livebook/hubs/settings.ex

```diff
@@ -0,0 +1,96 @@
+defmodule Livebook.Hubs.Settings do
+  @moduledoc false
+
+  alias Livebook.Storage
+  alias Livebook.Hubs.Hub
+
+  defmodule NotFoundError do
+    @moduledoc false
+
+    defexception [:id, plug_status: 404]
+
+    def message(%{id: id}) do
+      "could not find a hub matching #{inspect(id)}"
+    end
+  end
+
+  @namespace :hub
+
+  @doc """
+  Gets a list of hubs from storage.
+  """
+  @spec fetch_hubs() :: list(Hub.t())
+  def fetch_hubs do
+    Storage.current().all(@namespace)
+  end
+
+  @doc """
+  Gets one hub from storage.
+
+  Raises `NotFoundError` if the hub does not exist.
+  """
+  @spec hub_by_id!(String.t()) :: Hub.t()
+  def hub_by_id!(id) do
+    case Storage.current().fetch(@namespace, id) do
+      :error -> raise NotFoundError, id: id
+      {:ok, fields} -> struct!(Hub, fields)
+    end
+  end
+
+  @doc """
+  Checks if hub already exists.
+  """
+  @spec hub_exists?(Hub.t()) :: boolean()
+  def hub_exists?(%Hub{id: id}) do
+    case Storage.current().fetch(@namespace, id) do
+      :error -> false
+      {:ok, _} -> true
+    end
+  end
+
+  @doc """
+  Saves a new hub to the configured ones.
+  """
+  @spec save_hub(Hub.t()) :: Hub.t()
+  def save_hub(hub) do
+    attributes =
+      hub
+      |> Map.from_struct()
+      |> Map.to_list()
+
+    :ok = Storage.current().insert(@namespace, hub.id, attributes)
+
+    hub
+  end
+
+  @doc """
+  Subscribes to updates in hubs information.
+
+  ## Messages
+
+    * `{:hubs_changed, hubs}`
+
+  """
+  @spec subscribe(Livebook.Users.User.id()) :: :ok | {:error, term()}
+  def subscribe(user_id) do
+    Phoenix.PubSub.subscribe(Livebook.PubSub, "hubs:#{user_id}")
+  end
```

Comment:
To clarify, what I meant is this:

```elixir
defmodule Livebook.Hubs do
  def subscribe() do
    Phoenix.PubSub.subscribe(Livebook.PubSub, "hubs")
  end
end
```

Similar to `Livebook.Sessions`.

And the protocol can stay as `Livebook.Hub` :)

Reply:
Ah, actually this module seems to be a good place because that's where we fetch from. So just remove the id from subscribe, and let's rename the protocol back to `Livebook.Hub`, because it makes sense as singular :)

Reply:
I've been thinking about this protocol, because if I add the `use Ecto.Schema`, Dialyzer keeps warning about the struct not being the correct one to use with changesets. So I dunno if we should keep as is or add a new module to handle the schema (`Livebook.Hubs.Hub`). WDYT?

Reply:
But, about the singular thing, yeah, `Livebook.Hub` makes much more sense

Reply:
It makes sense to keep the protocol separate, because we may want functions related to the struct, and we can't define those in a protocol.